jax/_src/pallas/core.py

```diff
@@ -126,22 +126,12 @@ class Blocked:
 IndexingMode = Union[Blocked, Unblocked]
 
 
-@dataclasses.dataclass(init=False, unsafe_hash=True)
+@dataclasses.dataclass(unsafe_hash=True)
 class BlockSpec:
-  index_map: Callable[..., Any] | None
-  block_shape: tuple[int | None, ...] | None
-  memory_space: Any
-  indexing_mode: IndexingMode
-
-  def __init__(self, index_map: Callable[..., Any] | None = None,
-               block_shape: tuple[int | None, ...] | None = None,
-               memory_space: Any = None, indexing_mode: IndexingMode = blocked):
-    self.index_map = index_map
-    if block_shape is not None and not isinstance(block_shape, tuple):
```

Comment:
Is this conversion logic remaining the same?

Reply:
It isn't, i.e. we will keep the type of `block_shape` as is, but note that anything but `tuple` is explicitly disallowed by type annotations.